dataframe/dataframe.go

```diff
@@ -887,15 +898,23 @@ func LoadRecords(records [][]string, options ...LoadOption) DataFrame {
 	if cfg.hasHeader && len(records) <= 1 {
 		return DataFrame{Err: fmt.Errorf("load records: empty DataFrame")}
 	}
+	if len(cfg.names) > len(records[0]) {
+		return DataFrame{Err: fmt.Errorf("load records: too many column names")}
+	}
 
 	// Extract headers
 	headers := make([]string, len(records[0]))
 	if cfg.hasHeader {
 		headers = records[0]
 		records = records[1:]
-	} else {
+	}
+	for i, name := range cfg.names {
+		headers[len(headers)-len(cfg.names)+i] = name
+	}
+	if !cfg.hasHeader {
```

Comment:
It does work fine without that check, but it will cost some additional cycles. So why not just check if it's needed and only call the function if there is a need for it?